sharon-sharonshin/hw_2/randomAddress.js

```diff
@@ -0,0 +1,29 @@
+/*
+Optional bonus work on conditionals and functions:
+
+    Write a script that can generate random addresses
+    As a first step, create arrays that contain dummy data for each of the following: street number, street name, city name, state name, and zip code
+    Your script should randomly select one item from each of these arrays and then use them to construct a random address
+    Each time you load or reload the page in a browser, your program should print a new randomly-generated address to the terminal. For example:
+    
+    34578 Dolphin Street, Wonka NY, 44506
+*/
+
+window.onload = randomAddress;
+ 
+var streetNumber = ["1389", "5599", "2257", "1250", "4538"]
+var streetName = ["Amador Village Circle", "Huntcrest Drive", "Jefferson Street", "Fern Creek Drive"]
+var cityName = ["Oakland", "Chicago", "Los Angeles", "Atlanta"]
+var stateName = ["GA", "IL", "CA", "NV"]
+var zipCode = ["94612", "60661", "30332"]
+ 
+function randomAddress() {
+    console.log(
+        streetNumber[Math.floor(Math.random() * streetNumber.length)] + " " +
+        streetName[Math.floor(Math.random() * streetName.length)] + ", " +
+        cityName[Math.floor(Math.random() * cityName.length)] + " " +
+        stateName[Math.floor(Math.random() * stateName.length)] + ", " +
+        zipCode[Math.floor(Math.random() * zipCode.length)]
```

Comment:
I'm noticing an awesome pattern with the code you write, Sharon. Fancy one-liners that contain a lot of logic all at once are handled exactly the same at runtime as code that has been broken up into parts that make sense for the humans reading it. This is good practice, and I encourage you to keep up with it. 👍